### PR TITLE
mahoganyhomes: porting the plugin

### PR DESCRIPTION
**Pull request by grian32**
hoping i didnt fuck up anything lmao

### CODE REVIEW

File: mahoganyhomes/src/main/java/net/runelite/client/plugins/mahoganyhomes/Home.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+		this.location = location;
+		this.ladders = ladders;
+	}
+
+	String getName()
+	{
+		return WordUtils.capitalize(name().toLowerCase());
+	}
+
+	private static final ImmutableSet<Integer> LADDERS;
```
Comment (Lucwousin) [inline mahoganyhomes/src/main/java/net/runelite/client/plugins/mahoganyhomes/Home.java:90]: ImmutableSet is deprecated, use Set.of

File: mahoganyhomes/src/main/java/net/runelite/client/plugins/mahoganyhomes/Hotspot.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	MAHOGANY_HOMES_HOTSPOT_7(10560,
+		ImmutableSet.of(39987, 39995, 40006, 40015, 40094, 40099, 40162, 40170, 40177, 40292, 40295)),
+	MAHOGANY_HOMES_HOTSPOT_8(10561,
+		ImmutableSet.of(39988, 39996, 40163, 40289, 40299)),
+	;
+
+	private final int varb;
+	private final ImmutableSet<Integer> objectIds;
+
+	private static final ImmutableMap<Integer, Hotspot> HOTSPOT_BY_OBJECT_ID;
```
Comment (Lucwousin) [inline mahoganyhomes/src/main/java/net/runelite/client/plugins/mahoganyhomes/Hotspot.java:58]: Same idea as before

**Review by Lucwousin — CHANGES_REQUESTED** [review]
tiny modernizr type things

File: mahoganyhomes/src/main/java/net/runelite/client/plugins/mahoganyhomes/Hotspot.java (hunk 2 above)
Comment (grian32) [reply mahoganyhomes/src/main/java/net/runelite/client/plugins/mahoganyhomes/Hotspot.java:58]: done i think

File: mahoganyhomes/src/main/java/net/runelite/client/plugins/mahoganyhomes/Home.java (hunk 1 above)
Comment (grian32) [reply mahoganyhomes/src/main/java/net/runelite/client/plugins/mahoganyhomes/Home.java]: done i think

**Comment by grian32** [comment]
from my understanding the devs are making their own
and since > 21 days ago
im pretty sure im right
gonna close this

**Pull request merged**